### PR TITLE
mark the support of foreign key as experimental

### PR DESCRIPTION
**Pull request by qiancai**
### What is changed, added or deleted? (Required)

As in the title, the foreign key feature still has some limitations. It is not recommended to use this feature in production environments.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v7.5 (TiDB 7.5 versions)
- [x] v7.4 (TiDB 7.4 versions)
- [x] v7.3 (TiDB 7.3 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [x] Might cause conflicts after applied to another branch

### CODE REVIEW

File: releases/release-6.6.0.md [hunk 1]
```diff
@@ -176,7 +176,7 @@ In v6.6.0-DMR, the key new features and improvements are as follows:
 
 ### SQL
 
-* Support MySQL-compatible foreign key constraints [#18209](https://github.com/pingcap/tidb/issues/18209) @[crazycs520](https://github.com/crazycs520)
+* Support MySQL-compatible foreign key constraints (experimental)[#18209](https://github.com/pingcap/tidb/issues/18209) @[crazycs520](https://github.com/crazycs520)
```
Comment (qiancai) [inline releases/release-6.6.0.md:179]:
```suggestion
* Support MySQL-compatible foreign key constraints (experimental) [#18209](https://github.com/pingcap/tidb/issues/18209) @[crazycs520](https://github.com/crazycs520)
```

**Comment by lilin90** [comment]
Note: Since 6.6 (starting to support foreign key), 7.0, 7.2 docs have been archived and no longer receive updates, we only update it for 7.1, 7.3, 7.4 docs.

**Review by Oreoxmt — APPROVED** [review]

**Comment by easonn7** [comment]
LGTM

**Review by crazycs520 — APPROVED** [review]

**Pull request merged**